@here/harp-materials/lib/ShaderChunks/ExtrusionChunks.ts

```diff
@@ -21,10 +21,9 @@ vExtrusionAxis = vec4(normalMatrix * extrusionAxis.xyz, extrusionAxis.w);
     extrusion_normal_fragment_begin: `
 #ifdef FLAT_SHADED
     vec3 normal = vec3(0.0);
-    if (vExtrusionAxis.w > 0.999999) {
+    if (vExtrusionAxis.w > 0.999999 && length(vExtrusionAxis.xyz) > 0.000001) {
```

Comment:
Please add a comment why. Nobody will understand that condition in a year.
How does this look like on globe? The initial purpose of this code was to use smooth shading for the roof of the extruded geometry to fix the artifacts in the geojson example but now you switch to flat shading again if there is no elevation.
Would be good to have dedicated image regression test for that. As far as I remember @zbigg was adding some for extruded geometry. Maybe you can sync with him to make sure this case is covered. Sorry for asking this to you. I should have asked @amsXYZ to add the tests together with his change. Maybe we can also just revert his change and let him redo it.

Reply:
I created another PR that reverts @amsXYZ change:
https://github.com/heremaps/harp.gl/pull/818